docs/components/Button/Web.stories.tsx

```diff
@@ -74,4 +82,322 @@ const FormTemplate: ComponentStory<typeof Button> = () => (
+      <Grid.Cell size={{ xs: 6 }}>
+        <Button
+          label={`${type} ${variation} Large Button`}
+          type={type}
+          variation={variation}
+          size="large"
+        />
+      </Grid.Cell>
+
+      <Grid.Cell size={{ xs: 6 }}>
+        <Button type={type} variation={variation} size="large">
+          <Button.Label size="large">{`${type} ${variation} Large Button`}</Button.Label>
+        </Button>
+      </Grid.Cell>
+      <Grid.Cell size={{ xs: 6 }}>
+        <Button
+          label={`${type} ${variation} Loading Button`}
+          type={type}
+          variation={variation}
+          loading={true}
+        />
+      </Grid.Cell>
+      <Grid.Cell size={{ xs: 6 }}>
+        <Button type={type} variation={variation} loading={true}>
+          <Button.Label>{`${type} ${variation} Loading Button`}</Button.Label>
+        </Button>
+      </Grid.Cell>
+      <Grid.Cell size={{ xs: 6 }}>
+        <Button
+          label={`${type} ${variation} Disabled Button`}
+          type={type}
+          variation={variation}
+          disabled={true}
+        />
+      </Grid.Cell>
+      <Grid.Cell size={{ xs: 6 }}>
+        <Button type={type} variation={variation} disabled={true}>
+          <Button.Label>{`${type} ${variation} Disabled Button`}</Button.Label>
+        </Button>
+      </Grid.Cell>
+      <Grid.Cell size={{ xs: 6 }}>
+        <Button
+          label={`${type} ${variation} Combined Small Props`}
+          type={type}
+          variation={variation}
+          icon="note"
+          size="small"
+        />
+      </Grid.Cell>
+      <Grid.Cell size={{ xs: 6 }}>
+        <Button type={type} variation={variation} size="small">
+          <Button.Icon icon="note" size="small" />
+          <Button.Label size="small">{`${type} ${variation} Combined Small Props`}</Button.Label>
+        </Button>
+      </Grid.Cell>
+      <Grid.Cell size={{ xs: 6 }}>
+        <Button
+          label={`${type} ${variation} Combined Small Props on Right`}
+          type={type}
+          iconOnRight={true}
+          variation={variation}
+          icon="note"
+          size="small"
+        />
+      </Grid.Cell>
+      <Grid.Cell size={{ xs: 6 }}>
+        <Button type={type} variation={variation} size="small">
+          <Button.Label size="small">{`${type} ${variation} Combined Small Props on Right`}</Button.Label>
+          <Button.Icon icon="note" size="small" />
+        </Button>
+      </Grid.Cell>
+      <Grid.Cell size={{ xs: 6 }}>
+        <Button
+          label={`${type} ${variation} Combined Large Props`}
+          type={type}
+          variation={variation}
+          icon="note"
+          size="large"
+        />
+      </Grid.Cell>
+      <Grid.Cell size={{ xs: 6 }}>
+        <Button type={type} variation={variation} size="large">
+          <Button.Icon icon="note" size="large" />
+          <Button.Label size="large">{`${type} ${variation} Combined Large Props`}</Button.Label>
```

Comment:
I was debating on moving the size prop to a provider since it is used across the sub components but it was the only prop that was shared so I decided again it. I am not attached either way.

Reply:
Label and icon size feels like something we could at least default to the decision made at the parent level? There's a bit of nuance in the current Button in that the label only gets smaller in the small button, but does not get larger in the large, while the icon 's size is 1:1 with the button's size prop:
![image](https://github.com/user-attachments/assets/2cc23926-5ae5-4ff3-afe4-cd8f6607cad6)